sdk/core/Azure.Core/src/SyncAsyncEventHandler.cs

```diff
@@ -0,0 +1,217 @@
+    ///
+    /// try
+    /// {
+    ///     client.Snooze();
+    /// }
+    /// catch (AggregateException ex)
+    /// {
+    ///     ex.Handle(e =&gt; e is InvalidOperationException);
+    ///     Console.WriteLine(&quot;Please switch to your backup alarm.&quot;);
+    /// }
+    /// </code>
+    /// </exception>
+    /// <remarks>
+    /// <para>
+    /// Most Azure client libraries for .NET offer both synchronous and
+    /// asynchronous methods for calling Azure services.  You can distinguish
+    /// the asynchronous methods by their Async suffix.  For example,
+    /// BlobClient.Download and BlobClient.DownloadAsync make the same
+    /// underlying REST call and only differ in whether they block.  We
+    /// recommend using our async methods for new applications, but there are
+    /// perfectly valid cases for using sync methods as well.  These dual
+    /// method invocation semantics address the needs of our customers, but
+    /// require a little extra care when writing event handlers.
+    /// </para>
+    /// <para>
+    /// The SyncAsyncEventHandler is a delegate used by events in Azure client
+    /// libraries to represent an event handler that can be invoked from either
+    /// sync or async code paths.  It takes event arguments deriving from
+    /// <see cref="SyncAsyncEventArgs"/> that contain important information for
+    /// writing your event handler:
+    /// <list type="bullet">
+    /// <item>
+    /// <description>
+    /// <see cref="SyncAsyncEventArgs.CancellationToken"/> is a cancellation
+    /// token related to the original operation that raised the event.  It's
+    /// important for your handler to pass this token along to any asynchronous
+    /// or long-running synchronous operations that take a token so cancellation
+    /// (via something like
+    /// <c>new CancellationTokenSource(TimeSpan.FromSeconds(10)).Token</c>,
+    /// for example) will correctly propagate.
+    /// </description>
+    /// </item>
+    /// <item>
+    /// <description>
+    /// <see cref="SyncAsyncEventArgs.RunSynchronously"/> is a flag indicating
+    /// whether your handler was invoked synchronously or asynchronously.  If
+    /// you're calling sync methods on your client, you should use sync methods
+    /// to implement your event handler (you can return
+    /// <see cref="Task.CompletedTask"/>).  If you're calling async methods on
+    /// your client, you should use async methods where possible to implement
+    /// your event handler.  If you're not in control of how the client will be
+    /// used or want to write safer code, you should check the
+    /// <see cref="SyncAsyncEventArgs.RunSynchronously"/> property and call
+    /// either sync or async methods as directed.
+    /// </description>
+    /// </item>
+    /// <item>
+    /// <description>
+    /// Most events will customize the event data by deriving from
+    /// <see cref="SyncAsyncEventArgs"/> and including details about what
```

Comment:
```suggestion
    /// Most events will customize the event data by deriving from
    /// <see cref="SyncAsyncEventArgs"/> and include details about what
```

Reply:
"Deriving and Including" describe how we "Customize", rather than "Customize by deriving" and "Include details" being separate things.  Does that just read too awkwardly?

Reply:
The sentence structure was a bit awkward at first read, but I see now that it is correct.